src/fs.rs

```diff
@@ -112,6 +114,31 @@ pub fn iter_python_files<'a>(
         })
 }
 
+/// Create tree set with codes matching the pattern/code pairs.
+///
+/// Guarantees a `Some` result will be non-empty
+pub fn ignores_from_path(
+    path: &Path,
+    pattern_code_pairs: &[PerFileIgnore],
+) -> Result<Option<BTreeSet<CheckCode>>> {
+    let (file_path, file_basename) = extract_path_names(path)?;
+    let code_set = {
+        let mut rv = BTreeSet::new();
+        pattern_code_pairs.iter().for_each(|x| {
+            if is_excluded(file_path, file_basename, &[x.pattern.clone()]) {
+                rv.insert(x.code.clone());
+            }
+        });
+        rv
+    };
+    let rv = if code_set.is_empty() {
+        None
+    } else {
+        Some(code_set)
+    };
```

Comment:
I think I'd prefer to just return `Result<BTreeSet<CheckCode>>` and have the client check `.is_empty`. Using `None` to represent the empty state creates ambiguity in the typing (though I know you commented on it above).

Reply:
(That also lets you get rid of `code_set` entirely and just return `rv`.)